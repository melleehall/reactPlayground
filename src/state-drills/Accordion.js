import React from 'react';
import './Accordion.css';

export default class Accordion extends React.Component {
    static defaultProps = {
        sections: []
    }

    state = {
        currentViewIndex: null,
    }

    handleButtonClick(index) {
        this.setState({currentViewIndex: index})
    }

    renderButtons() {
        return this.props.sections.map((section, index) => (
            <li className='Accordion__item' key={index}>
                <button 
                    type='button'
                    onClick={() => this.handleButtonClick(index)}
                >
                    {section.title}
                </button>
                {(this.state.currentViewIndex === index) && <p>{section.content}</p>}
            </li>
        ))
      }

    render () {
        return (
            <ul className='Accordion'>
                {this.renderButtons()}
            </ul>
        )
    }
}