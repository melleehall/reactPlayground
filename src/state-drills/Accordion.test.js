import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordion from './Accordion';

// an array of objects each containing two keys, 'title' and 'content'
const sectionsProp = [
    {
      title: 'Section 1',
      content: 'ONE Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'TWO Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'THREE Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]

describe('Accordion Component', () => {
    it('renders without errors', () => {
        const div = document.createElement('div')
        ReactDOM.render(<Accordion />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders empty list item given no sections', () => {
        const wrapper = shallow(<Accordion />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    // it('renders no sections as active by default', () => {
    //     const wrapper = shallow(<Accordion sections={sectionsProp} />)
    //     expect(toJson(wrapper)).toMatchSnapshot()
    // })
})