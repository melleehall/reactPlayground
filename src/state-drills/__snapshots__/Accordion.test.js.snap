// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component renders empty list item given no sections 1`] = `
<ul
  className="Accordion"
/>
`;

exports[`Accordion Component renders no sections as active by default 1`] = `
<ul
  className="Accordion"
>
  <li
    className="Accordion__item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="Accordion__item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    className="Accordion__item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
  <p
    className="Accordion__item"
  >
    ONE Lorem ipsum dolor sit amet consectetur adipisicing elit.
  </p>
</ul>
`;
